workflow/executor.go

```diff
@@ -792,7 +809,16 @@ func (e *executor) prepareDependencies( //nolint:gocognit,gocyclo
 	case reflect.Struct:
 		switch s := stepData.(type) {
 		case expressions.Expression:
-			dependencies, err := s.Dependencies(outputSchema, workflowContext)
+			// Evaluate the dependencies of the expression on the data structure.
+			// Include extraneous is false due to this only being needed for connecting to the appropriate node,
+			// as opposed to the data provided by that node.
+			pathUnpackRequirements := expressions.UnpackRequirements{
+				ExcludeDataRootPaths:     false,
+				ExcludeFunctionRootPaths: true, // We don't need to setup DAG connections for them.
+				StopAtTerminals:          true, // I don't think we need the extra info. We just need the connection.
```

Comment:
"I don't think we need" is not a particularly encouraging code comment! Can we say something definite here?

Reply:
Fixed.